.evergreen/evergreen.yml.in

```diff
@@ -19,13 +21,20 @@ const NODE_VERSIONS = [
   { shortName: '14', versionSpec: NODE_JS_VERSION_14, skipNodeVersionCheck: '' }
 ];
 
+const pathToPackages = path.join(__dirname, '..', 'packages');
+const MONGOSH_PACKAGES = fs.readdirSync(pathToPackages, { withFileTypes: true })
+  .filter(d => d.isDirectory() && fs.existsSync(path.join(pathToPackages, d.name, 'package.json')))
+  .map(d => d.name);
```

Comment:
Two ideas here … Firstly, would it maybe make sense to add a flag to the package.json files that indicates that the package does not run integration tests against mongod, and so it’s sufficient to only add one entry per (package × Node.js version)? Only `cli-repl`, `java-shell`, `node-runtime-worker-thread`, `service-provider-server` and `shell-api` actually run integration tests, so that would help with cutting the matrix down quite significantly.

And secondly, maybe we could add another field that indicates an array of build variants where the tests should run; So that for `java-shell` and `browser-repl`, we don’t even generate tasks for the variants where they are no-ops anyway. That would also allow us to get rid of the `run-if-platform` hack, and for the `mongosh` package that array could just be empty.

Reply:
That sounds pretty neat to me! Should we add a `mongosh` key to the `package.json` and then put those config options in there?

Reply:
Sure :) I don’t think we really need to worry about naming collisions here anyway